master/buildbot/data/builds.py

```diff
@@ -100,12 +102,31 @@ def startConsuming(self, callback, options, kwargs):
                                                  ('builds', str(buildid), None))
 
     def control(self, action, args, kwargs):
-        if action != "stop":
+        # we convert the action into a mixedCase method name
+        action_method = getattr(self, "action" + action[0].upper() + action[1:])
```

Comment:
FWIW, your code is fine, but you can do "action[0].capitalize()" for the same effect.